### ISSUE TITLE
Guidance for Multibranch Pipelines in Hygieia

### ISSUE THREAD

**Issue by splatthy** [Hygieia/Hygieia#3192]
I'm looking for guidance on how to integrate multi-branch pipelines with Github and Jenkins into a Hygieia Team Dashboard (and ultimately roll that information up to a Product dashboard).

As an example, I have development teams working within Github on multiple Release branches.  So for example they'll be working on two branches in parallel called release/january and release/february, there could also be a hotfix branch specific for hotfix code in Prod.

Right now we point Hygieia's SCM integration with Github at the repo and a specific branch, there didn't seem to be any way that I saw to allow it to collect information across the repo.

We use the Jenkins plugin to push build, and artifact information during CI to Hygieia.  These build pipelines are triggered on git commit as multibranch looking for specific string matches in the branches.  So on the team dashboard, when I try to add the build in for these multibranch pipelines and search for the job name, it looks something like: **Jenkins-CloudBees : job/<Job Folder>/job/<JobName>/job/<branch name>/**.  Which means each dashboard would map to one specific code branch.

If the branches were static, this wouldn't be such a big deal, we could just create team dashboards for each branch, and while messy get the desired outcome in a roundabout way.  But with the teams likely building release branches named after the relative release windows, and merging/deleting old releases over time, this would get pretty messy.

Is this configuration something Hygieia can support, or intended as a future enhancement?

**Comment by rvema** [comment]
@splatthy currently on the UI there is a limitation of displaying the SCM widget for only 1 combination of url and branch.

**Comment by splatthy** [comment]
Thanks @rvema, this slipped and I accidentally let this go stale.  I meant to follow up and ask if there's anything in the roadmap that might remove the limitation?  On our end it's not a pressing need now, but over the next few months we're going to be enabling more multi-branch CI.

Knowing that we're generating builds from multiple branches, is there anything that comes to mind to visualize that without having to have dedicated branches?  Perhaps if we directly fed the commit data into the mongodb during the Jenkins build job?  Not ideal obviously, but we could maintain within the mongodb (just the db, not the repo itself) a "dummy" branch that we seed with commit info during the build, of all release branches that we build from.  Then use that to drive the visuals.  I'm thinking of ideas here that are probably duct tape and string, but just trying to figure out options to support the use case.

**Comment by rvema** [comment]
not sure why this was marked stale, I will keep this open.

**Comment by Ayush-Goyal-coding** [comment]
Hi,
Wouldn't there be a problem with multi branch Configuration , for other widgets for e.g. 
sonar and fortify would have the scan results of one branch only.

**Comment by rvema** [comment]
The data model and apis support the multi branch pipeline , limitation is only in UI for displaying the data. The UI limitation will be addressed as part of Widget redesign effort scopes for later part of this year

**Comment by Ayush-Goyal-coding** [comment]
Yeah I totally agree with that part, I was just talking on a conceptual level
Suppose if we have multiple widgets of repo and Jenkins maybe one point to dev branch and another to master
Maybe in dev branch, the code coverage is increased and the number of issues is reduced, so for sonar data, it wouldn't be clear for which branch the data is represented in the dashboard.
This is just my thought, I might be wrong too... 

Thanks for the reply @rvema
